codalab/apps/web/templates/web/admin_competitions_manager.html

```diff
@@ -4,154 +4,336 @@
+                                method: 'POST',
+                                headers: {
+                                    'X-CSRFToken': csrftoken,
+                                    'Content-Type': 'application/json',
+                                },
+                                body: JSON.stringify(competitionsJson)
+                            })
+                            .then(response => {
+                                if(response.ok) {
+                                    return response.json();
+                                }
+                                throw new Error('Something went wrong');
+                            })
+                            .then(json => {
+                                this.competitionsIdToUpdate = [];
+                                for (const competition of json) {
+                                    const index = this.competitionsTableItems.findIndex(comp => comp.id == competition.id);
+                                    if (index >= 0) {
+                                        // Using method splice for Vue reactivity (updating an element is not supported otherwise)
+                                        this.competitionsTableItems.splice(index, 1, competition);
+                                    } else {
+                                        console.log("error: No matching competition to update");
+                                    }
+                                }
+                            })
+                            .catch(error => {
+                                console.log("error while fetching /api/admin/competitions", error);
+                            })
+                            .finally(() => {
+                                this.saving = false;
+                            });
+                        }
+                    },
+                    synchronize(competition) {
+                        this.synchronizing = true;
+                        fetch("/api/admin/competition/" + competition.id + "/apply_upper_bound_limit", {
+                            method: 'PATCH',
+                            headers: {
+                                'X-CSRFToken': csrftoken,
+                            }
+                        })
+                        .then(response => {
+                            if(response.ok) {
+                                return response.json();
+                            }
+                            throw new Error('Something went wrong');
+                        })
+                        .then(competition => {
+                            const index = this.competitionsTableItems.findIndex(comp => comp.id == competition.id);
+                            if (index >= 0) {
+                                // Using method splice for Vue reactivity (updating an element is not supported otherwise)
+                                this.competitionsTableItems.splice(index, 1, competition);
```

Comment:
In this case splice is acting exactly like the following line of code: `this.competitionsTableItems[index] = competition`
I had so many issues about Vue reacticity that I am just using splice without thinking.

**Quick explanation:**
Vue reactivity works with reference.
That means: if you have a custom component that takes some props, and you update those props in the parent component then the only way the child data gets updated is that you have actually updated the reference to the props.

Working example:
```js
parent.vue
<template>
  <custom-child :title="title"/>
</template>
<script>
  data: {
    title: "some-text"
  }
</script>

child.vue
<template>
  <p>{{title}}</p>
</template>
<script>
  props: ['title']
  data: {}
</script>
```
If you update the title in the parent component by doing `this.title = "other-text"` then in the child the displayed title will be updated. Because you are assigning a new object (so a new reference) to a reactive property (this.title)

But it does not work with Arrays:
Not working example:
```js
parent.vue
<template>
  <custom-child :titles="titles"/>
</template>
<script>
  data: {
    titles: ["title1", "title2", "title3"]
  }
</script>

child.vue
<template>
  <li v-for="title in titles" :key="title">
    {{ title }}
  </li>
</template>
<script>
  props: ['titles']
  data: {}
</script>
```
If you update one value in the array using classic assignment in the parent component: `this.titles[1] = "new title 2"` then you actually don't change the reference you are passing to the child component so Vue will not update the child component. In order to make it work you have to create a new object. And for arrays one easy way to do it is to use the `Array.splice` method.

If you want more in-depth understanding you can check the [official Vue website](https://v2.vuejs.org/v2/guide/reactivity.html#For-Arrays)

__In conclusion__:
I haven't tried to just use basic assignment. May be it will work. But I had so many nightmares about that that I am just using splice now ^^